### PR TITLE
Change Edit Profile route

### PR DESCRIPTION
**Pull request by bonnichiwa**
## Link to Issue
Closes: #[3173](https://github.com/hicommonwealth/commonwealth/issues/3173)

## Description of Changes
- Change `/profile/:id/edit` to `/profile/edit`
- Will automatically render the profile that is of the logged in user to edit
- When not logged in and accessing `/profile/edit` prompt user to log in and render edit profile page upon logging in

Screen Shot 2023-03-17 at 4 25 40 PM

Screen Shot 2023-03-17 at 4 26 56 PM

### CODE REVIEW

File: packages/commonwealth/client/scripts/router.ts
Comment (masvelio) [inline packages/commonwealth/client/scripts/router.ts:1]: this file is no-op. This should go to customDomainRoutes.tsx

**Review by masvelio — CHANGES_REQUESTED** [review]
comment

File: packages/commonwealth/server/routes/getNewProfile.ts [hunk 1]
```diff
@@ -32,13 +32,21 @@ const getNewProfile = async (
...
   const { profileId } = req.query;
-  if (!profileId) return next(new Error(Errors.NoIdentifierProvided));
+  let profile;
 
-  const profile = await models.Profile.findOne({
-    where: {
-      id: profileId,
-    },
-  });
+  if (profileId) {
```
Comment (jnaviask) [inline packages/commonwealth/server/routes/getNewProfile.ts:37]: Is this ideal behavior? Ideally we should require the GET route to pass a profile id, even if a user is logged in.

File: packages/commonwealth/client/scripts/router.ts [hunk 2]
```diff
@@ -318,13 +318,10 @@ const getCustomDomainRoutes = (importRoute) => ({
...
   }),
-  '/profile/id/:profileId/edit': importRoute(
-    import('views/pages/edit_new_profile'),
-    {
-      scoped: true,
-      deferChain: true,
-    }
-  ),
+  '/profile/edit': importRoute(import('views/pages/edit_new_profile'), {
+    scoped: true,
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/router.ts:322]: These pages are not scoped.

**Review by jnaviask — APPROVED** [review]
A few small comments, but generally OK pending QA (given we want to get this out ASAP).

File: packages/commonwealth/client/scripts/router.ts
Comment (CowMuon) [reply packages/commonwealth/client/scripts/router.ts:1]: What needs to happen to merge master into react once this goes in? @jnaviask @masvelio

**Review by CowMuon — APPROVED** [review]

**Review by jnaviask — APPROVED** [review]

**Pull request merged**